modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/UserResourceManagerTest.java

```diff
@@ -115,89 +364,865 @@ public void testGetSuccess(String id, String attributes,
+        Mockito.when(userManager.listUsersWithPost(any(SearchRequest.class),
+                anyMap())).thenThrow(new CharonException());
+        SCIMResponse scimResponse = userResourceManager.listWithPOST(RESOURCE_STRING, userManager);
+        Assert.assertEquals(scimResponse.getResponseStatus(), ResponseCodeConstants.CODE_INTERNAL_ERROR);
+    }
+
+    @Test
+    public void testListWithPOSTNotImplementedException()
+            throws NotImplementedException, BadRequestException, CharonException {
+
+        abstractResourceManager.when(() -> AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_USER_ENDPOINT + "/.search");
+        abstractResourceManager.when(()
+                -> AbstractResourceManager.encodeSCIMException(any(NotImplementedException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new NotImplementedException()));
+        Mockito.when(userManager.listUsersWithPost(any(SearchRequest.class),
+                anyMap())).thenThrow(new NotImplementedException());
+        SCIMResponse scimResponse = userResourceManager.listWithPOST(RESOURCE_STRING, userManager);
+        Assert.assertEquals(scimResponse.getResponseStatus(), ResponseCodeConstants.CODE_NOT_IMPLEMENTED);
+    }
+
+    @Test
+    public void testListWithPOSTBadRequestException()
+            throws NotImplementedException, BadRequestException, CharonException {
+
+        abstractResourceManager.when(() -> AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_USER_ENDPOINT + "/.search");
+        abstractResourceManager.when(()
+                -> AbstractResourceManager.encodeSCIMException(any(BadRequestException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new BadRequestException()));
+        Mockito.when(userManager.listUsersWithPost(any(SearchRequest.class),
+                anyMap())).thenThrow(new BadRequestException());
+        SCIMResponse scimResponse = userResourceManager.listWithPOST(RESOURCE_STRING, userManager);
+        Assert.assertEquals(scimResponse.getResponseStatus(), ResponseCodeConstants.CODE_BAD_REQUEST);
+    }
+
+    @DataProvider(name = "dataForUpdateWithPATCH")
+    public Object[][] dataToUpdateWithPATCH() throws BadRequestException, CharonException, InternalErrorException {
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+        JSONDecoder decoder = new JSONDecoder();
+        User userOld = decoder.decodeResource(NEW_USER_SCIM_OBJECT_STRING_FOR_PATCH, schema, new User());
+        String id = userOld.getId();
+        User userNew = decoder.decodeResource(NEW_USER_SCIM_OBJECT_STRING_FOR_PATCH_UPDATE, schema, new User());
+        return new Object[][]{
+                {id, NEW_USER_SCIM_OBJECT_STRING_FOR_PATCH_UPDATE, "userName", null, userNew, userOld}
```

Comment:
can include more datasets considering different combinations of values for `attributes` and `excludeAttributes`

Reply:
Fixed in commit 504369f8bdb4f024c63ae5d78f16b04decbedc11